src/workerd/api/node/buffer.c++

```diff
@@ -85,30 +88,20 @@ void SwapBytes(kj::ArrayPtr<kj::byte> bytes) {
   }
 }
 
-enum class Encoding {
-  ASCII,
-  LATIN1,
-  UTF8,
-  UTF16LE,
-  BASE64,
-  BASE64URL,
-  HEX,
-};
-
-Encoding getEncoding(kj::StringPtr encoding) {
-  if (encoding == "utf8"_kj) {
+Encoding getEncoding(kj::StringPtr input) {
+  if (input == "utf8"_kj) {
     return Encoding::UTF8;
-  } else if (encoding == "ascii") {
+  } else if (input == "ascii"_kj) {
     return Encoding::ASCII;
-  } else if (encoding == "latin1") {
+  } else if (input == "latin1"_kj) {
     return Encoding::LATIN1;
-  } else if (encoding == "utf16le") {
+  } else if (input == "utf16le"_kj) {
     return Encoding::UTF16LE;
-  } else if (encoding == "base64") {
+  } else if (input == "base64"_kj) {
     return Encoding::BASE64;
-  } else if (encoding == "base64url") {
+  } else if (input == "base64url"_kj) {
     return Encoding::BASE64URL;
-  } else if (encoding == "hex") {
+  } else if (input == "hex"_kj) {
     return Encoding::HEX;
   }
 
```

Comment:
This isn't new in this PR but I've been thinking that we should change this `KJ_UNREACHABLE` into a proper `JSG_FAIL_REQUIRE` error instead. This value should be checked on the JS side before we get here but if it isn't, then it will lead to a crash.

So instead of `KJ_UNREACHABLE` at the end of this let's make it ...

```
JSG_FAIL_REQUIRE(Error, kj::str("Invalid encoding: ", input));
```

Or something similar.

Reply:
I agree that an error is better than crashing!

I'm fine with updating this in here. And additional PR for this small change will not benefit anyone. I'm updating it now. Thanks for the review.